recipe-server/client/control_new/utils/apiFetch.js

```diff
@@ -36,5 +36,28 @@ export default async function apiFetch(url, options = {}) {
   }
 
   const response = await fetch(`${API_ROOT}${url}${queryString}`, settings);
+
+  // Throw if we get a non-200 response.
+  if (!response.ok) {
+    let message;
+    let data;
+
+    try {
+      data = await response.json();
+      message = data.detail || response.statusText;
+    } catch (error) {
+      message = error.message;
+    }
+
+    throw new ApiError(message, data);
+  }
   return response.json();
 }
+
+export class ApiError extends Error {
```

Comment:
nit, but I'd move this to the top of the document since it's short and is used within this same file.